apps/web/src/scenes/CommunityPage/CommunityPageOwnershipRequiredModal.tsx

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import { Button } from '~/components/core/Button/Button';
+import { HStack, VStack } from '~/components/core/Spacer/Stack';
+import { BaseM } from '~/components/core/Text/Text';
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import { CommunityPageOwnershipRequiredModalFragment$key } from '~/generated/CommunityPageOwnershipRequiredModalFragment.graphql';
+import useSyncTokens from '~/hooks/api/tokens/useSyncTokens';
+import { RefreshIcon } from '~/icons/RefreshIcon';
+
+type Props = {
+  communityRef: CommunityPageOwnershipRequiredModalFragment$key;
+  refetchIsMemberOfCommunity: () => void;
+};
+
+export default function CommunityPageOwnershipRequiredModal({
+  communityRef,
+  refetchIsMemberOfCommunity,
+}: Props) {
+  const community = useFragment(
+    graphql`
+      fragment CommunityPageOwnershipRequiredModalFragment on Community {
+        name
+        chain
+      }
+    `,
+    communityRef
+  );
+
+  const { hideModal } = useModalActions();
+  const handleOkClick = useCallback(() => {
+    hideModal();
+  }, [hideModal]);
+  const { syncTokens } = useSyncTokens();
+
+  // Acknowledgment: This could let a user refresh a chain they can't access in the Editor
+  const handleRefreshCollectionClick = useCallback(async () => {
+    if (!community.chain) {
+      return;
+    }
+    setIsRefreshing(true);
+    await syncTokens(community.chain);
+    setIsRefreshing(false);
+    refetchIsMemberOfCommunity();
+    hideModal();
+  }, [community.chain, hideModal, refetchIsMemberOfCommunity, syncTokens]);
+
+  const [isRefreshing, setIsRefreshing] = useState(false);
```

Comment:
nit: I think we have `isLocked` in the `useSyncTokens` hooks. Not really sure should we used the local state or state in the hooks 🤔

Reply:
updated